[UG] Add annotated GUI info and format student fields table (#317)
AY2122S1-CS2103T-F13-3/tp: docs/UserGuide.md

<kbd>text</kbd> | A keyboard input, or a button to be clicked on.
[text](#about-this-guide) | Links to other parts of the document, or links to be opened in the browser.
_text_ | Caption for images.
<div markdown="block" class="alert alert-info"> :information_source: **Note** </div> | Indication that the following text is a note, which is useful in helping you understand how TAB works.

> <div markdown="span" class="alert alert-primary"> :bulb: **Tip** </div> | Indication that the following text is a tip, which can allows you to interact with TAB more effectively and quickly. 

Reviewer: ```suggestion
<div markdown="span" class="alert alert-primary"> :bulb: **Tip** </div> | Indication that the following text is a tip, which can allow you to interact with TAB more effectively and quickly. 
```

This section briefly explains the various section of TAB's GUI.

![layout](images/annotatedGui.png)
<div class="caption">Basic layout of TAB's user interface.</div>

Section | Representation
---------|---------
Menu Bar | The area where you can click to navigate and view some of TAB's features (e.g. help, reminder and calendar). See [Features](#features) for more details.
Menu Item | A button that you can click to view the stated feature, or to quit TAB (for <kbd>Exit</kbd>).

> Command Box | The text field where you key in commands that are meant to be processed by TAB. TAB will execute the command after you pressed <kbd>Enter</kbd>.

Reviewer: ```suggestion
Command Box | The text field where you key in commands that are meant to be processed by TAB. TAB will execute the command after you press <kbd>Enter</kbd>.
```

> Result Display | The area that shows the result of the execution of the command. If the command entered has been executed successfully, it will display relevant success messages, otherwise, it will show error messages indicating the cause of the error.

Reviewer: ```suggestion
Result Display | The area that shows the result of the execution of the command. If the command entered has been executed successfully, it will display the relevant success message. Otherwise, it will show an error message indicating the cause of the error.
```

Student List | The area that shows the list of students you have in TAB.

> Student Card | The area that shows the details of a student. See [Managing Students](#managing-students) for more details.

Reviewer: Area for card sounds strange but I don't know what could be a better alternative

Lesson List | The area that shows the name of the student you have selected together with a list of lessons that the student has.

> Lesson Card | The area that shows the details of a lesson. See [Managing Lessons](#managing-lessons) for more details.

Reviewer: Same comments for these.

![studentCard](images/annotatedStudentCard.png)

<div class="caption">Layout of a student card.</div>

<div markdown="block" class="alert alert-info">
**:information_source: Note:**<br>
  <ul>
    <li>A student can be identified by the student index number shown in the displayed list of students.</li>
    <li>Fields that are empty will not be displayed.</li>
  </ul>
</div>

<table id="student-param-table">
    <thead>
        <tr>
            <th style="text-align:center; padding: 10px">Category</th>
            <th style="text-align:center">Field</th>
            <th style="text-align:center">Prefix</th>
            <th style="text-align:center">Constraint(s)</th>
            <th style="text-align:center">Example(s)</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td rowspan=2>Essential</td>
            <td>Name</td>
            <td><code>n/</code></td>
            <td>
              <ul> 
                <li>Must be present.</li>
                <li>Case-sensitive.</li>
                <li>TAB does not allow adding students with the same case-sensitive name.</li>
              </ul>
            </td>

>             <td><code>n/Jane</code></td>

Reviewer 2: ```suggestion
            <td><code>n/Jane Tan</code></td>
```

            <td><code>stream/</code></td>
            <td>Not Applicable</td>
            <td><ul>
              <li><code>stream/A-Level</code></li>
              <li><code>stream/</code> clears the academic stream field, if there were any.</li>
            </ul></td>
        </tr>
        <tr>
            <td>Academic Level</td>
            <td><code>lvl/</code></td>
            <td><ul>
              <li>Maximum 15 characters, including space(s).</li>
            </ul></td>
            <td><ul>
              <li><code>lvl/J1</code></li>
              <li><code>lvl/</code> clears the academic level field, if there were any.</li>
            </ul></td>
        </tr>
        <tr>
            <td>Remark</td>
            <td><code>r/</code></td>
            <td>Not Applicable.</td>
            <td><ul>
              <li><code>r/She is weak at maclaurin series.</code></li>
              <li><code>r/</code> clears the remark field, if there were any.</li>
            </ul></td>
        </tr>
        <tr>
            <td>Tags</td>
            <td><code>t/</code></td>
            <td><ul>
              <li>Must be alphanumeric characters.</li>
              <li>Case-insensitive</li>

>               <li>To add multiple tags to a student, you need to add <code>t/</code> before every tag name.</li>

Reviewer 2: I think adding multiple tags should be address in ladd and not here

Reviewer 2: Just need these examples I think
`t/unpaid` is a valid tag
`t/unpaid retained` is invalid

Reviewer 2: can remove the multiple tags one, since that depends on the command whether it accepts multiple prefixes

This section guides you on how to use the commands for managing students in TAB.
Executing any of the commands in this section will bring you to the students interface. Any command from other sections, such as `day` or `tag`, will bring you right out.

![studentCard](images/annotatedStudentCard.png)

<div class="caption">Layout of a student card.</div>

<div markdown="block" class="alert alert-info">
**:information_source: Note:**<br>
  <ul>
    <li>A student can be identified by the student index number shown in the displayed list of students.</li>
    <li>Fields that are empty will not be displayed.</li>
  </ul>
</div>

<table id="student-param-table">
    <thead>
        <tr>
            <th style="text-align:center; padding: 10px">Category</th>
            <th style="text-align:center">Field</th>
            <th style="text-align:center">Prefix</th>

>             <th style="text-align:center">Constraint(s)</th>
>             <th style="text-align:center">Example(s)</th>

Reviewer 2: It's fine without
```suggestion
            <th style="text-align:center">Constraints</th>
            <th style="text-align:center">Examples</th>
```

        </tr>
    </thead>
    <tbody>
        <tr>
            <td rowspan=2>Essential</td>
            <td>Name</td>
            <td><code>n/</code></td>
            <td>
              <ul> 

>                 <li>Must be present.</li>

Reviewer 2: It's inferred from essential
```suggestion
```

                <li>Case-sensitive.</li>
                <li>TAB does not allow adding students with the same case-sensitive name.</li>
                <li>If there are multiple name inputs prefixed with <code>n/</code>, TAB only accepts the last name input.</li>
              </ul>
            </td>
            <td><ul>
              <li><code>n/Jane</code></li>
              <li><code>n/Jane n/John</code> - name will be <code>John</code>.</li>
            </ul></td>
        </tr>
        <tr>
            <td>Address</td>
            <td><code>a/</code></td>
            <td><ul>

>               <li>Must be present.</li>

Reviewer 2: ```suggestion
```

              <li>If there are multiple address inputs prefixed with <code>a/</code>, TAB only accepts the last address input.</li>
            </ul></td>
            <td><ul>
              <li><code>a/Blk 123A, Serangoon Ave 3, #04-56</code></li>
              <li><code>a/Blk 123A, Serangoon Ave 3, #04-56 a/Clementi</code> - address will be <code>Clementi</code>.</li>
            </ul></td>
        </tr>
        <tr>
            <td rowspan=4 id="student-contact">Contact</td>
            <td id="phone-field">Phone</td>
            <td><code>p/</code></td>
            <td><ul>

>               <li><strong>At least one</strong> of the contact fields must be present.</li>

Reviewer 2: Add in the notes block above instead?

            <td><ul>
              <li><code>r/She is weak at maclaurin series.</code></li>
              <li><code>r/weak at series r/weak at trigo</code> - remark will be <code>weak at trigo</code></li>
              <li><code>r/</code> clears the remark field, if there were any.</li>
            </ul></td>
        </tr>
        <tr>
            <td>Tags</td>
            <td><code>t/</code></td>
            <td><ul>
              <li>Must be alphanumeric characters.</li>
              <li>Case-insensitive</li>
              <li>To add multiple tags to a student, you need to add <code>t/</code> before every tag name.</li>
            </ul></td>
            <td><ul>
              <li><code>t/unpaid</code> adds only one tag with tag name "UNPAID"</li>
              <li><code>t/unpaid retained</code> is invalid.</li>
              <li><code>t/unpaid t/retained</code> adds two tags with tag names "UNPAID" and "RETAINED"</li>
              <li><code>t/</code> clears <strong>all</strong> tags, if there were any.</li>
            </ul></td>
        </tr>
        <tr>
            <td rowspan="2">View Only</td>
            <td>Student Index</td>

>             <td>Not Applicable.</td>

Reviewer 2: no need full stop
```suggestion
            <td>Not Applicable</td>
```

A student can have any number of tags (including 0).
</div>

> * **At least one** [contact](#student-contact) field is required.<br>

Reviewer 2: make the link [contact field]()

#### Viewing upcoming lessons : `remind`

Displays a list of upcoming lessons that ends within the next 48 hours.
- Lesson cards in the reminder list do not display the date range and the cancelled dates, if any.

Format: `remind`

![remind](images/remind.png)

<div class="caption">Reminder window interface.</div>

Example: Suppose the date today is 1 Nov 2021 and current time is 1500h, 
- lessons with the following dates and time are considered upcoming:
  - 1 Nov 2021 with end time at or after 1500h, 
  - 2 Nov 2021 with any valid time range, 
  - 3 Nov 2021 with start time before or at 1500h.
- lessons with the following dates and time are not considered upcoming:
  - dates before 1 Nov 2021 (has passed),
  - 1 Nov 2021 with end time before 1500h (has passed),

>   - 3 Nov 2021 with start time after 1500 (beyond 48 hours).

Reviewer 2: ```suggestion
  - 3 Nov 2021 with start time after 1500h (beyond 48 hours).
```

TAB has keyboard shortcuts for navigating between its various views to help you stay more productive by keeping your hands on the keyboard.

Shortcut | Action | Description
--------|---------|---------
<kbd>F1</kbd> | Help | Opens the help window.
<kbd>F2</kbd> | View Students | Brings you to the last displayed list of students. <br> e.g. if you used `find` to filter the list of students, and switched to the calendar view, pressing <kbd>F2</kbd> brings you back to the filtered list.
<kbd>F3</kbd> | View Calendar | Brings you to the calendar interface.
<kbd>F4</kbd> | View Tags | Brings you to your list of tags.

> <kbd>F5</kbd> | View Reminder | Opens the reminder window with the list of upcoming lessons with end date time within the next 48 hours.

Reviewer: Not sure if end date time is used by non-technical person.
```suggestion
<kbd>F5</kbd> | View Reminder | Opens the reminder window with the list of upcoming lessons that ends within the next 48 hours.
```
